front/src/applications/editor/tools/switchEdition/components/TrackNodeTypeDiagram.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { TRACK_NODE_TYPES_ID, TrackNodeTypeId } from 'types';
+import link from 'assets/pictures/nodeTypes/link.svg';
+import crossing from 'assets/pictures/nodeTypes/crossing.svg';
+import pointSwitch from 'assets/pictures/nodeTypes/point_switch.svg';
+import singleSlipSwitch from 'assets/pictures/nodeTypes/single_slip_switch.svg';
+import doubleSlipSwitch from 'assets/pictures/nodeTypes/double_slip_switch.svg';
+
+const TrackNodeTypeDiagram = ({ trackNodeType }: { trackNodeType: TrackNodeTypeId }) => {
+  const trackNodeTypeImage = () => {
```

Comment:
Use a const here, it will be easier to manipulate
```suggestion
  const TrackNodeTypeImage = useMemo(() => {...}, [trackNodeType])
```

Reply:
Done